mongomock/collection.py

```diff
@@ -277,7 +277,8 @@ def execute(self, write_concern=None):
     def add_insert(self, doc):
         self.insert(doc)
 
-    def add_update(self, selector, doc, multi, upsert, collation=None):
+    def add_update(self, selector, doc, multi, upsert, collation=None,
+                   array_filters=None):
```

Comment:
This change is not related to my fix, however the tests with pymongo lib present would fail if this wasn't there. I can split it in a different commit/PR if you prefer.

Reply:
What does `array_filters` do on real pymongo/mongodb?

Reply:
https://docs.mongodb.com/manual/reference/method/Bulk.find.arrayFilters/
It determines which array elements to modify for an update operation on an array field.

Reply:
So it's a bit problematic to mock it without doing anything. It means there is some use case that would work differently on mongomock and on the real implementation...

Reply:
I see. I thought it was OK, as it's exactly what happens with the collation arg that is not used either. I found the discussion in https://github.com/mongomock/mongomock/pull/393 which explains why we don't want to add it.

I'm dropping it.

I still find puzzling that tests do not pass...

Reply:
Yes, I agree there are a lot of confusing things in how Pymongo affects environments.